test/common/upstream/load_balancer_fuzz_base.cc

```diff
@@ -0,0 +1,118 @@
+
+namespace Random {
+uint64_t FakeRandomGenerator::random() {
+  uint8_t index_of_data = counter % bytestring_.size();
+  ++counter;
+  ENVOY_LOG_MISC(trace, "random() returned: {}", bytestring_.at(index_of_data));
+  return bytestring_.at(index_of_data);
+}
+} // namespace Random
+
+namespace Upstream {
+
+// Anonymous namespace for helper functions
+namespace {
+std::vector<uint64_t>
+constructByteVectorForRandom(const Protobuf::RepeatedField<uint64_t>& byte_string) {
+  std::vector<uint64_t> byteVector;
+  byteVector.reserve(byte_string.size());
+  for (const auto eightBytes : byte_string) {
+    byteVector.push_back(eightBytes);
+  }
+  return byteVector;
+}
+} // namespace
+
+void LoadBalancerFuzzBase::initializeFixedHostSets(uint32_t num_hosts_in_priority_set,
+                                                   uint32_t num_hosts_in_failover_set) {
+  int port = 80;
+  for (uint32_t i = 0; i < num_hosts_in_priority_set; ++i) {
+    host_set_.hosts_.push_back(makeTestHost(info_, "tcp://127.0.0.1:" + std::to_string(port)));
+    ++port;
+  }
+  for (uint32_t i = 0; i < num_hosts_in_failover_set; ++i) {
+    failover_host_set_.hosts_.push_back(
+        makeTestHost(info_, "tcp://127.0.0.1:" + std::to_string(port)));
+    ++port;
+  }
+  // TODO(zasweq): More than two host sets?
+}
+
+// Initializes random and fixed host sets
+void LoadBalancerFuzzBase::initializeLbComponents(
+    const test::common::upstream::LoadBalancerTestCase& input) {
+  random_.bytestring_ = constructByteVectorForRandom(input.bytestring_for_random_calls());
+  initializeFixedHostSets(input.num_hosts_in_priority_set(), input.num_hosts_in_failover_set());
+}
+
+// Updating host sets is shared amongst all the load balancer tests. Since logically, we're just
+// setting the mock priority set to have certain values, and all load balancers interface with host
+// sets and their health statuses, this action maps to all load balancers.
+void LoadBalancerFuzzBase::updateHealthFlagsForAHostSet(bool failover_host_set,
+                                                        uint32_t num_healthy_hosts,
+                                                        uint32_t num_degraded_hosts,
+                                                        uint32_t num_excluded_hosts) {
+  MockHostSet& host_set = *priority_set_.getMockHostSet(int(failover_host_set));
+  // Will not overflow because size is capped by port numbers
+  uint32_t host_set_size = host_set.hosts_.size();
+  uint32_t i = 0;
+  for (; i < std::min(num_healthy_hosts, host_set_size); ++i) {
+    host_set.healthy_hosts_.push_back(host_set.hosts_[i]);
+  }
+  for (; i < std::min(num_healthy_hosts + num_degraded_hosts, host_set_size); ++i) {
+    host_set.degraded_hosts_.push_back(host_set.hosts_[i]);
+  }
+
+  for (; i < std::min(num_healthy_hosts + num_degraded_hosts + num_excluded_hosts, host_set_size);
+       ++i) {
+    host_set.excluded_hosts_.push_back(host_set.hosts_[i]);
+  }
+
+  host_set.runCallbacks({}, {});
```

Comment:
Could there be a host in two (or more) vectors when there are multiple calls to `updateHealthFlagsForAHostSet`? I'm not sure if it's possible, but consider the following scenario:
1. `updateHealthFlagsForAHostSet(1,1,1);` - which adds a single host to the healthy, degraded and excluded vectors.
1. `updateHealthFlagsForAHostSet(1,2,0);` - which adds a single host to the healthy, and two hosts to degraded, but host_set.hosts_[2] is now in both the degraded and excluded vectors.
Should these lists be cleared when `updateHealthFlagsForAHostSet` is called?

Reply:
Hey Adi, thanks for the pass! First, I will be switching this to a subset selection per Harvey's comment. Yup, you caught a bug I found last night (luckily I'll be changing this method anyway!) where I forgot to clear the healthy, degraded, and excluded hosts, so each vector would keep growing arbitrarily. Will change with next commit.

Reply:
@zasweq GH review tip; if a thread is effectively resolved, best to "Resolve conversation" to make it easier for reviewers :)